### DISCUSSION TITLE
Is there a way to include the query string in the short url?

### DISCUSSION THREAD

**Discussion by markjr** [YOURLS/YOURLS#3505] [Q&A]
This is a great suite of software.

Our use case is we're using this to repair a boatload of broken backlinks to our old help pages, which over the years have had various different subdomains ("web", "rtfm", "helpwiki", etc)

Example, we have a lot of broken backlinks from **fusion.easydns.com** to fix, most of them are in the format:

https://fusion.easydns.com/Knowledgebase/Article/View/190/0/aname-records

So we can just set the short url to **/Knowledgebase/Article/View/190/0/aname-records** and redirect to the relevant new page on our site.

But I just came across a URL under this same subdomain, and here is the problem:

https://fusion.easydns.com/index.php?/Knowledgebase/Article/View/190/7/aname-records/

We can't just use /index.php for the shorturl - somehow we need to be able to key on the entire URI _with the query string_

I figured this wasn't as easy as just copying the allow-hyphens plugin and changing it to an underscore (something I already did and will upload when this is all wrapped up). But I did test it and no dice.

The various append query string plugins aren't quite what we need - because we need to be able to select from the database, based on what the query string is.

I'm willing to code this and will happily contribute it here when it's working, I just need some guidance on what to do. Looking at append_qs plugin it looks like $_SERVER['QUERY_STRING'] is visible, I need to know where to modify the database insertion on the short url and where to key on it in the redirect loader.

I hope that makes sense.

(Alternatively, if somebody who's actually familiar and experienced with this is up for it, we'll happily hire you to code this and then contribute it to the project once it's working).

**Comment by ozh** [comment]
Not sure I entirely understand your problem, in particular how you're using YOURLS to fix 404.

Anyway: plugins such as "use hyphens" are used to allow hyphens in the short URL, not in the long URL. There is no problem with shortening `/index.php?/Knowledgebase/Article/View/190/7/aname-records/` : example on https://ozh.in/easydns

**Reply by markjr** [reply]
I'm talking about the short URL.

In our use case:

short url = old, broken link on a subdomain or old domain
long url = new URL that works

We point the DNS for the hostname of the old urls to the Yourls server, use the short urls to capture the broken links, 301 them to the current pages.

Yes I know the use hyphens, etc apply to the short url - (old urls) those are what we're trying to match.

So your test case is reversed, we need "/index.php?/Knowledgebase/Article/View/190/7/aname-records/" as the short url (remember, short doesn't mean _short_ in our use case, it means _old_) and https://ozh.in/easydns as the long.

Make sense?

there's actually an entire use case here for exactly this sort of situation.

**Reply by dgw** [reply]
Much as we always appreciate when people use YOURLS, I don't think it's the best tool for this use case. I suspect that it would be much easier to accomplish this with rewrites at the httpd level.

**Reply by markjr** [reply]
> Much as we always appreciate when people use YOURLS, I don't think it's the best tool for this use case. I suspect that it would be much easier to accomplish this with rewrites at the httpd level.

For URLs under the same domain as the redirect, sure. But the analytics, API, and the rest of the feature set, it's actually a good fit. Especially for migrating between domains.
